airbyte-config/models/src/main/java/io/airbyte/config/storage/DefaultGcsClientFactory.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.config.storage;
+
+import com.google.api.client.util.Preconditions;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageOptions;
+import io.airbyte.config.storage.CloudStorageConfigs.GcsConfig;
+import java.util.function.Supplier;
+
+/**
+ * Takes in the constructor our standard format for gcs configuration and provides a factory that
+ * uses that configuration to create a GCS client (Storage).
+ */
+public class DefaultGcsClientFactory implements Supplier<Storage> {
```

Comment:
What if we didn't use the factory pattern and directly injected the configurations into the Client classes?

I might be wrong - don't really see the value of the Factory here since there isn't any sort of dynamic runtime class creation that the Factory pattern is good for.

This would remove some LOCs and remove an abstraction layer.

Reply:
Yeah. You're right, it's not obvious why we need it in this PR, but I plan to use these factories for the worker state storage (my next PR). So this is set up for that.

Reply:
Ok!